library/spdm_requester_lib/libspdm_req_get_certificate.c

```diff
@@ -158,11 +160,28 @@ return_status try_spdm_get_certificate(IN void *context, IN uint8_t slot_id,
             status = RETURN_DEVICE_ERROR;
             goto done;
         }
+        if (spdm_request.offset == 0) {
+            total_responder_cert_chain_buffer_length = spdm_response.portion_length + spdm_response.remainder_length;
+        }
+        else if (spdm_request.offset + spdm_response.portion_length + spdm_response.remainder_length != total_responder_cert_chain_buffer_length) {
+            status = RETURN_DEVICE_ERROR;
+            goto done;
+        }
+
+        if (spdm_request.offset == 0) {
```

Comment:
Do we still need this logic?

Line 163~169 already covered below check.

Reply:
Yes, the logic  `else if (spdm_request.offset + spdm_response.portion_length + spdm_response.remainder_length != total_responder_cert_chain_buffer_length)` is strong and already covers the check: `if (remainder_length[n] < remainder_length[n - 1])` .

And I have changed the code. Thanks.